_examples/json/main.go

```diff
@@ -0,0 +1,86 @@
+// nolint: golint, dupl
+package main
+
+import (
+	"os"
+
+	"github.com/alecthomas/kong"
+
+	"github.com/alecthomas/participle/v2"
+	"github.com/alecthomas/participle/v2/lexer"
+)
+
+var (
+	jsonLexer = lexer.MustSimple([]lexer.SimpleRule{
+		{Name: "Comment", Pattern: `\/\/[^\n]*`},
+		{Name: "String", Pattern: `"(\\"|[^"])*"`},
+		{Name: "Number", Pattern: `[-+]?(\d*\.)?\d+`},
```

Comment:
This is not 100% accurate, eg. it won't match `1e10`, but it's okay for an example.

Reply:
I am not sure whether json support scientific notation

Reply:
It definitely does.